tools/ingest_historical_csv.py

```diff
@@ -92,15 +92,16 @@ def _dedupe_data(df: pd.DataFrame) -> pd.DataFrame:
 
 
 def _update_df(df: pd.DataFrame) -> pd.DataFrame:
-    df['ecosystem'] = df['ecosystem'].str.upper()
-    df['status'] = df.apply(lambda x: _get_status_type(x['status']), axis=1)
+    if len(df) != 0:
+        df['ecosystem'] = df['ecosystem'].str.upper()
+        df['status'] = df.apply(lambda x: _get_status_type(x['status']), axis=1)
 
-    if 'cve_model_flag' not in df:
-        df['probable_cve'] = True
-    else:
-        df['probable_cve'] = df.apply(lambda x: _get_probabled_cve(x['cve_model_flag']), axis=1)
+        if 'cve_model_flag' not in df:
+            df['probable_cve'] = True
+        else:
+            df['probable_cve'] = df.apply(lambda x: _get_probabled_cve(x['cve_model_flag']), axis=1)
```

Comment:
This can also be done directly be done with df.apply, the function you pass here doesn't necessary have to be a lambda and you can check in the function itself if the 'cve_model_flag' property exists.

It's a matter of preference so no need to change it but just FYI.